src/main/java/catan/services/impl/PlayServiceImpl.java

```diff
@@ -125,17 +131,26 @@ private void endTurn(GameBean game) throws GameException {
             Integer newPreparationCycle = game.getPreparationCycle();
             boolean preparationFinished = newPreparationCycle == null && !game.getCurrentMove().equals(1);
             shouldUpdateNextMove = (preparationFinished || previousPreparationCycle.equals(newPreparationCycle));
+        } else {
+            game.setDiceThrown(false);
+            game.setDiceFirstValue(null);
+            game.setDiceSecondValue(null);
         }
 
         if (shouldUpdateNextMove) {
             playUtil.updateNextMove(game);
-            //TODO: set it only in MAIN stage
-            game.setDiceThrown(false);
         }
     }
 
     private void throwDice(GameBean game) {
+        Integer diceFirstValue = randomUtil.pullRandomDiceNumber(new ArrayList<Integer>(asList(1, 2, 3, 4, 5, 6)));
+        Integer diceSecondValue = randomUtil.pullRandomDiceNumber(new ArrayList<Integer>(asList(1, 2, 3, 4, 5, 6)));
+        game.setDiceFirstValue(diceFirstValue);
+        game.setDiceSecondValue(diceSecondValue);
         game.setDiceThrown(true);
+        if (diceFirstValue + diceSecondValue != 7) {
+            mainStageUtil.distributeResources(game);
```

Comment:
'distribute' sound like it is something related to goods trading) maybe change this word to "propagate"? or leave it as is...